plugins/catalog-backend/src/next/DefaultProcessingStateManager.ts

```diff
@@ -38,37 +46,24 @@ export class DefaultProcessingStateManager implements ProcessingStateManager {
     });
   }
 
-  async addProcessingItems(request: AddProcessingItemRequest) {
-    return this.db.transaction(async tx => {
-      await this.db.addUnprocessedEntities(tx, request);
-    });
-  }
-
-  async getNextProcessingItem(): Promise<ProccessingItem> {
-    const entities = await new Promise<RefreshStateItem[]>(resolve =>
-      this.popFromQueue(resolve),
-    );
-    const { id, state, unprocessedEntity } = entities[0];
-    return {
-      id,
-      entity: unprocessedEntity,
-      state,
-    };
-  }
-
-  async popFromQueue(resolve: (rows: RefreshStateItem[]) => void) {
-    const entities = await this.db.transaction(async tx => {
-      return this.db.getProcessableEntities(tx, {
-        processBatchSize: 1,
+  async getNextProcessingItem(): Promise<ProcessingItem> {
```

Comment:
Wondering if it makes sense to have multiple engines running simultaneously, and have this fetch a larger batch of items and cache them internally. Not a huge batch, just perhaps the same size as the number of engines. Only running a single engine per instance is prolly gonna be too slow. Ofc we could totally make the engine be able to process stuff in parallel too, but that's pretty much the same thing.

Reply:
Or starting N engines per process (especially if we start breaking out the loops as separate worker threads)

Reply:
(where each of them fetches one work item at a time, to be clear)